presto-main/src/main/java/com/facebook/presto/metadata/Metadata.java

```diff
@@ -114,6 +116,29 @@
     Map<QualifiedObjectName, List<ColumnMetadata>> listTableColumns(Session session, QualifiedTablePrefix prefix);
 
     /**
+     * Gets the TableIdentity for the specified table.
+     */
+    @NotNull
+    TableIdentity getTableIdentity(Session session, TableHandle tableHandle);
+
+    /**
+     * Gets the ColumnIdentity for all columns of the specified table.
+     */
+    @NotNull
+    Map<String, ColumnIdentity> getColumnIdentities(Session session, TableHandle tableHandle);
```

Comment:
Why not just have `ColumnIdentity getColumnIdentitie(Session session, ColumnHandle columnHandle)`